### ISSUE TITLE
Fitness Function Explanation

### ISSUE THREAD

**Issue by tomatopuree** [ultralytics/yolov5#5093]
## Question

From the docs:

Fitness is the value we seek to maximize. In YOLOv5 we have defined a default fitness function as a weighted combination of metrics: mAP@0.5 contributes 10% of the weight and mAP@0.5:0.95 contributes the remaining 90%. You may adjust these as you see fit or use the default fitness definition.

```
def fitness(x): 
     # Model fitness as a weighted combination of metrics 
     w = [0.0, 0.0, 0.1, 0.9]  # weights for [P, R, mAP@0.5, mAP@0.5:0.95] 
     return (x[:, :4] * w).sum(1) 
```

My question:

What does the @0.5 mean at the end of mAP@0.5. Similarly what does the @0.5:0.95 mean at the end of mAP@0.5:0.95.

Also what does P and R stand for in the fitness function?

**Comment by glenn-jocher** [comment]
@tomatopuree you can learn about object detection metrics here: 
https://towardsdatascience.com/map-mean-average-precision-might-confuse-you-5956f1bfa9e2

**Comment by tomatopuree** [comment]
I see about the numbers following map.

This still does not explain what P and R stand for. Is it precision and recall? I would suggest not leaving this unclear in the tutorial.

**Comment by glenn-jocher** [comment]
@tomatopuree thanks for the suggestion! Can you link directly to the URL you'd like to see updated?

**Comment by tomatopuree** [comment]
https://docs.ultralytics.com/yolov5/tutorials/hyperparameter_evolution

Under 2. Define Fitness, an explanation of the notation of numbers after @ and what P and R exactly stand for can be useful for clarity.

Are you confirming that P and R stand for Precision and Recall?

**Comment by glenn-jocher** [comment]
@tomatopuree oh got it! Yes P and R are precision and recall. Wikipedia has a good article on them:
https://en.wikipedia.org/wiki/Precision_and_recall

**Comment by glenn-jocher** [comment]
@tomatopuree #607 has been updated per your recommendations! Let us know if there are any other improvements you might want to see.

**Comment by tomatopuree** [comment]
Hey Glen,

I do know what the concepts are; just didn't want to assume what shorthand maps to what concept. The additions you've made to the tutorial are great.

And I must say that this is a great repo you have here. From all the augmentations to transfer learning to the built in graphing, this is an awesome project. You are the Keras of the real time object detection paradigm.

My regards

**Comment by ZachL1** [comment]
hi, @glenn-jocher ,thanks for your great work!

I have a question about the weights used when calculating fitness. In the default implementation, why the weights for precision and recall are set to 0, is there any reason? It seems to me that these two metrics should be important in real applications.

**Comment by glenn-jocher** [comment]
@ZachL1 hi there! Thank you for your kind words and for reaching out with your question.

Regarding the weights for precision and recall set to 0 in the default implementation of the fitness function, the reason is that in YOLOv5, the focus is primarily on the mAP (mean Average Precision) metric. While precision and recall are certainly important metrics in object detection, the default fitness function in YOLOv5 prioritizes mAP.

That being said, you are always free to adjust the weights in the fitness function according to your specific requirements and priorities. Feel free to experiment and customize the weights based on your application needs.

I hope this helps clarify things for you! Please let me know if you have any further questions.

**Comment by ZachL1** [comment]
I see, thank you for such a quick reply!

**Comment by glenn-jocher** [comment]
@ZachL1 you're welcome! If you have any more questions, feel free to ask.
